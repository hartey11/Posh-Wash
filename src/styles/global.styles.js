import { css } from "@emotion/core"
import normalize from "normalize.css"

const globalStyles = ({ isMenuOpen, scrollY, theme }) => css`
  ${normalize};
  :root {
    --swiper-theme-color: #fff;
  }

  html {
    box-sizing: border-box;
  }

  *,
  *:before,
  *:after {
    box-sizing: inherit;
  }

  body {
    ${theme.body};
    background-color: #1c1c1c;
    color: #fff;
    font-family: ${theme.fontFamily.body};
    left: ${isMenuOpen && `0`};
    position: ${isMenuOpen && `fixed`};
    right: ${isMenuOpen && `0`};
    top: ${isMenuOpen && `-${scrollY}px`};
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    ${theme.heading};
  }

  h1 {
    ${theme.h1}
  }
  h2 {
    ${theme.h2}
  }
  h3 {
    ${theme.h3}
  }
  h4 {
    ${theme.h4}
  }
  h5 {
    ${theme.h5}
  }
  h6 {
    ${theme.h6}
  }

  p,
  ol,
  ul,
  pre {
    ${theme.content}
  }

  a {
    color: inherit;
  }

  #gatsby-focus-wrapper {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
  }

  .swiper-pagination-bullets {
    display: flex;
    justify-content: center;
  }

  .swiper-pagination-bullet {
    border-radius: 0;
    display: block;
    height: 10px;
    width: 10px;

    &:not(.swiper-pagination-bullet-active) {
      background-color: #fff;
      opacity: 0.4;
    }
  }

  .button {
    ${theme.heading};
    border: 2px solid #fff;
    padding: ${theme.baseSpacing(0.5)} ${theme.baseSpacing()};
    text-align: center;
    text-decoration: none;
  }
`

export default globalStyles
