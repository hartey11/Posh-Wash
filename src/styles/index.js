export { default as globalStyles } from "./global.styles"
export { default as mediaQuery } from "./media-query"
export { default as theme } from "./theme"
