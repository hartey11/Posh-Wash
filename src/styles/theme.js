import Shevy from "shevyjs"

const shevy = new Shevy({ proximity: true })

const theme = {
  ...shevy,
  breakpoints: {
    "mobile-md": 360,
    "mobile-lg": 480,
    "tablet-sm": 600,
    "tablet-md": 800,
    desktop: 1025,
    "widescreen-sm": 1366,
    "widescreen-md": 1440,
    "widescreen-lg": 1600,
    hd: 1920,
    "ultra-hd": 3840,
    "4k": 4096,
    "retina-5k": 5120,
  },
  fontFamily: {
    heading: "Orbitron, sans-serif",
    body: "Montserrat, sans-serif",
  },
  header: {
    borderWidth: 2,
    height: {
      mobile: 50,
      tablet: 50,
      desktop: 60,
    },
    spacer(breakpoint) {
      return this.height[breakpoint]
    },
  },
  heading: {
    get fontFamily() {
      return theme.fontFamily.heading
    },
    fontWeight: 700,
    letterSpacing: "0.1em",
    textTransform: "uppercase",
  },
  maxWidth: 1300,
}

export default theme
