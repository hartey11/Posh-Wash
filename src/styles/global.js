import { css } from "@emotion/core"
import normalize from "normalize.css"

import theme from "./theme"

const global = css`
  ${normalize};
  html {
    box-sizing: border-box;
  }

  *,
  *:before,
  *:after {
    box-sizing: inherit;
  }

  body {
    ${theme.body};
    font-family: ${theme.fontFamily.body};
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    ${theme.heading};
  }

  h1 {
    ${theme.h1}
  }
  h2 {
    ${theme.h2}
  }
  h3 {
    ${theme.h3}
  }
  h4 {
    ${theme.h4}
  }
  h5 {
    ${theme.h5}
  }
  h6 {
    ${theme.h6}
  }

  p,
  ol,
  ul,
  pre {
    ${theme.content}
  }

  #gatsby-focus-wrapper {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
  }
`

export default global
