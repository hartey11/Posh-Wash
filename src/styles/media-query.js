/**
 * TODO: Convert to NPM package.
 * @desc
 * @if string assume min width
 * @if object use key, value provided with and operator
 *
 * @param object/string query
 * @param bool/string max-width - only used if query is a string.
 *
 * @return string - @media ${queryString}

 * @examples
 * min-width: mediaQuery("mobile-lg");
 * max-width: mediaQuery("mobile-lg", true);
 * min and max: mediaQuery("mobile-md", "desktop");
 * custom query: mediaQuery({ "min-width": "mobile-lg", "max-width": "mobile-lg", orientation: "portrait" });
 */

import theme from "./theme"

const { breakpoints } = theme

const isMaxWidthQuery = max => {
  return typeof max === "boolean" && max
}

const getBreakpointValue = (name, maxWidth = false) => {
  if (Object.prototype.hasOwnProperty.call(breakpoints, name)) {
    return `${
      isMaxWidthQuery(maxWidth) ? breakpoints[name] - 0.02 : breakpoints[name]
    }px`
  }
  return name
}

const addFeature = (query, maxWidth = false) => {
  const key = `${isMaxWidthQuery(maxWidth) ? "max" : "min"}-width`
  const value = getBreakpointValue(query, maxWidth)

  return {
    [key]: value,
  }
}

const mediaQuery = (query, maxWidth = false) => {
  let features = {}

  if (typeof query === "object") {
    features = { ...query }
  }

  if (typeof query === "string") {
    if (typeof maxWidth === "boolean") {
      features = addFeature(query, maxWidth)
    }

    if (typeof maxWidth === "string") {
      features = addFeature(query, false)
      Object.assign(features, addFeature(maxWidth, true))
    }
  }

  const queryString = Object.entries(features).reduce(
    (accumulator, [name, value], i) => {
      const string = `(${name}: ${getBreakpointValue(value)})`

      if (i === 0) {
        return string
      }

      return `${accumulator} and ${string}`
    },
    ""
  )

  return `@media ${queryString}`
}

export default mediaQuery
