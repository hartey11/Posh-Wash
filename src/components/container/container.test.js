import React from "react"
import { ThemeProvider } from "emotion-theming"
import { render, cleanup } from "@testing-library/react"
import Container from "./container.component"

import { theme } from "../../styles"

afterEach(cleanup)

describe("Container component", () => {
  it("should render correctly in the browser", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Container columns={2}>
          <div>Column 1</div>
          <div>Column 2</div>
        </Container>
      </ThemeProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
