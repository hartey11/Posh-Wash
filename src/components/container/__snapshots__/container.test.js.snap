// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component should render correctly in the browser 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-left: auto;
  margin-right: auto;
  max-width: 800px;
  padding: 30.599999999999998px 20.4px;
}

@media (orientation:portrait) {
  .emotion-0 {
    max-width: 600px;
  }
}

@media (min-width:800px) {
  .emotion-0 {
    padding: 51px 20.4px;
  }
}

@media (min-width:1025px) {
  .emotion-0 {
    max-width: 1300px;
    padding: 71.39999999999999px 40.8px;
  }
}

@media (min-width:1366px) {
  .emotion-0 {
    padding: 81.6px 40.8px;
  }
}

.emotion-0 > * {
  border: 2px solid #fff;
  margin: 10.2px 0;
  padding: 20.4px;
  width: 100%;
}

@media (orientation:landscape) {
  .emotion-0 > * {
    width: calc(50% - 10.2px);
  }
}

@media (min-width:1025px) {
  .emotion-0 > * {
    width: calc(50% - 10.2px);
  }
}

@media (min-width:1366px) {
  .emotion-0 > * {
    margin: 20.4px 0;
    padding: 40.8px;
    width: calc(50% - 20.4px);
  }
}

.emotion-0 > * > *:last-child {
  margin-bottom: 0;
}

.emotion-0 h1 {
  font-size: clamp(20px,6.25vw,24px);
}

.emotion-0 h1:first-of-type {
  margin-top: 0;
}

.emotion-0 h2 {
  font-size: clamp(16px,5vw,20px);
}

<div
    class="emotion-0"
  >
    <div>
      Column 1
    </div>
    <div>
      Column 2
    </div>
  </div>
</DocumentFragment>
`;
